### ISSUE TITLE
Multi-AZ support for aws_fsx_windows_file_system

### ISSUE THREAD

**Issue by pmcevoy** [terraform-providers/terraform-provider-aws#11246]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In November 2019, AWS announced Multi-AZ support for Amazon FSx for Windows.  Can the `aws_fsx_windows_file_system` be updated to support this?

### New or Affected Resource(s)

* aws_fsx_windows_file_system

### Potential Terraform Configuration
The existing terraform config syntax does not necessarily need to change - just remove the precondition on `subnet_ids` containing only 1 subnet id.  Multi-AZ can be indicated by providing 2 subnet Ids
```hcl
resource "aws_fsx_windows_file_system" "example" {
  active_directory_id = "${aws_directory_service_directory.example.id}"
  kms_key_id          = "${aws_kms_key.example.arn}"
  storage_capacity    = 300
  subnet_ids          = ["${aws_subnet.example.id}", "${aws_subnet.example2.id}"]
  throughput_capacity = 1024
}
```

### References
* https://aws.amazon.com/about-aws/whats-new/2019/11/amazon-fsx-for-windows-file-server-now-supports-file-systems-that-span-multiple-availability-zones/
* https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystem.html#FSx-CreateFileSystem-request-SubnetIds
> For Windows MULTI_AZ_1 file system deployment types, provide exactly two subnet IDs, one for the preferred file server and one for the standy file server.

**Comment by mrgiles** [comment]
@pmcevoy, From what I'm reading in the API documentation, it will require a new option (boolean) for **DeploymentType** and an IF condition, if DeploymentType = Multi-AZ, then enter a value for **PreferredSubnetId**. so not so simple to implement. The API has changed.

https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystemWindowsConfiguration.html#FSx-Type-CreateFileSystemWindowsConfiguration-DeploymentType

https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystemWindowsConfiguration.html#FSx-Type-CreateFileSystemWindowsConfiguration-PreferredSubnetId

**Comment by pmcevoy** [comment]
Surely Multi-AZ is implied when more than 1 id is supplied in the `subnet_ids` list?  As for preferred, couldn't that be simply the first element in that list?

**Comment by mrgiles** [comment]
From what I see, both in the Console and the API, there's this new option **DeploymentType** that takes a string value of MULTI_AZ_1 | SINGLE_AZ_1, with the later being the default. When MULTI_AZ_1 is selected, another new option has to be completed, **PreferredSubnetId**. 

See Request Syntax:
https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystem.html#API_CreateFileSystem_RequestSyntax

You have `"SubnetIds": [ "string" ]` at the top level (default, single AZ) and, under the `"WindowsConfiguration" {}` block, you have `"DeploymentType": "string"` () and `"PreferredSubnetId": "string"`.

That's why I think that additional code is required to address these 2 new objects, just listing 2 subnets in SubnetIds won't automatically convert the deployment to multi-AZ.

**Comment by pmcevoy** [comment]
Ah - what I mean is that the -resource provider- api does not need to change: certainly internally the implementation does need to pass those settings. Unfortunately I am no Go expert so cannot even attempt a fix..

**Comment by pmcevoy** [comment]
CloudFormation now supports MultiAz:

https://forums.aws.amazon.com/ann.jspa?annID=7378

**Comment by Calange** [comment]
The multi-az is supported by the AWS Go SDK : https://github.com/aws/aws-sdk-go/releases/tag/v1.29.9

**Comment by kth-xx** [comment]
Any update on support for multi-az?

**Comment by mgiles-sfdc** [comment]
@pmcevoy this issue has (finally) been solved by PR [#12676 ](https://github.com/terraform-providers/terraform-provider-aws/pull/12676)

Thank you @DrFaust92! And thank you @breathingdust for reviewing & merging!

We've been waiting for this functionality since Dec'19

**Comment by mrgiles** [comment]
@pmcevoy this issue has (finally) been solved by [PR#12676](https://github.com/terraform-providers/terraform-provider-aws/pull/12676)

Thank you @DrFaust92! And thank you @breathingdust for approving & merging!

We've been waiting for this functionality since Dec'19

**Comment by bflad** [comment]
This has been released in [version 3.0.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/v3.0.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
